sparse_merkle_tree.js

```diff
@@ -68,12 +69,36 @@ class SparseMerkleTree {
     }
 
     async verify(root, queries, proof) {
-        return new Promise((resolve, reject) => {
-            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (err, result) => {
-                if (err) {
-                    reject(err);
-                    return;
+        return new Promise((resolve) => {
+            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (_, result) => {
+                resolve(result);
+            });
+        });
+    }
+
+    async verifyInclusionProof(root, queries, proof) {
+        return new Promise((resolve) => {
+            for (let i = 0; i < queries.length; i++) {
+                if (!isInclusion(queries[i], proof.queries[i])) {
+                    resolve(false);
                 }
+            }
+
+            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (_, result) => {
+                resolve(result);
+            });
+        });
+    }
+
+    async verifyNonInclusionProof(root, queries, proof) {
+        return new Promise((resolve) => {
+            for (let i = 0; i < queries.length; i++) {
+                if (isInclusion(queries[i], proof.queries[i])) {
```

Comment:
Didn't we mentioned (@ricott1) that there could be three possibilities?
Proof may be inclusion, non-inclusion or undefined?
Opposite of `inclusion` is not `nonInclusion` and wice versa.
So here we actually need to test `!isNonInclusion` and this function is than missing in `utils.js`?

Reply:
I guess it was for unit tests.

Reply:
That was for the specific function defined in the unit tests. The difference was that that function accepted a list of query keys as input, rather than a single key, hence there was the possibility of having some keys included and some not.